### ISSUE TITLE
Atted object passes quotation marks as part of var_name and att_name to subprocess

### ISSUE THREAD

**Issue by cemac-ccs** [nco/pynco#60]
This may be a 'me' problem, but I have found that when using the Atted object, the nco.ncatted command fails. This is, as far as I can see, due to the return value of the Atted object being formatted as  
```
return '-a "{0}","{1}",{2},{3},{4}'.format(
            self.att_name, self.var_name, self.mode, type_char, strvalue
        )
```
which results in an error message saying something along the lines of 'no variable named "time" found'. 

As an example, running the following script will fail:

```
from nco.custom import Atted
from nco import Nco

nco=Nco()
ifile = /path/to/input/file
ofile = /path/to/output/file

att_val = -999
var = "time"
att = "_FillValue"
att_type="i"

opts=[Atted(mode="modify",att_name=att,var_name=var,value=att_val,stype=att_type)]

nco.ncatted (input=ifile, output=ofile, options=opts)
```

The error message it will state that the call attempted would be 

```
/usr/bin/ncatted -a "_FillValue","time",m,i,-999 --output=/path/to/output/file /path/to/input/file
```
which if run directly in the terminal will execute correctly.

From what I can see, it seems that when the string is passed to subprocess.Popen, the quotation marks are preserved in the string, and so the variable name and attribute names are read by ncatted as including the quotation marks. I have confirmed this by trying to run the above string as

```
subprocess.run(ncatted_string.split())
```

which gives the same error.

If instead of using the Atted object I call nco.ncatted with the option string 

```
opts = [" -a {0},{1},m,{2},{3}".format(att, var, att_type, att_val)]
```

then the call works correctly, and so the issue seems to be only with the Atted object in nco.custom.

I cannot personally confirm whether this is also an issue with the Limit, LimitSingle or Rename objects as I have only been using ncatted, however a quick look at the prn_option functions in these classes shows quotation marks similar to those in the prn_option function for Atted, and so a similar error may occur for these also.

I encountered this error using pynco 1.0.0

**Comment by czender** [comment]
Your diagnosis may well be correct, and, if so, we would be happy to accept a patch. However, it is unclear to me whether your proposed solution works in cases when the attribute name or variable name contained whitespace. In the shell world, this is what the quotes preserve. If your solution does "the right thing" for names with whitespace, then no reason not to adopt it. @hmb1 may have an opinion, too.

**Comment by cemac-ccs** [comment]
A colleague has pointed out to me that if the command is invoked as 
```
opts=[Atted(mode="modify",att_name=att,var_name=var,value=att_val,stype=att_type)]

nco.ncatted (input=ifile, output=ofile, options=opts, use_shell=True)
```
then the quotes and whitespace should be passed to the shell and interpreted in the way you would want, rather than being escaped by subprocess and so the command is run correctly. A possible alternative solution then would be to have the default value of use_shell as True where this behaviour might be encountered

**Comment by hmb1** [comment]
@cemac-ccs 
 I am unable to recreate your issue. Your example script works for me.  
From the nco.py script I see that **use_shell**  already defaults to True on the line:  201
> use_shell = kwargs.pop("use_shell", True)

When I change the above to False I get your results

**Comment by cemac-ccs** [comment]
I see. It seems to be due to the fact that I am using the tagged 1.0.0 version of pynco. Looking at the nco.py file in my conda env, I see the default for use_shell is set to False in the get function, unlike what you can see on the main branch on the repo.

This will solve itself then once you put the next version up on conda forge, and in the meantime can be easily circumvented by adding a little note in the docs to say that for pynco 1.0.0 the Atted object requires `use_shell=True` to be set manually.

**Comment by mx-moth** [comment]
This issue will be resolved by #62

**Comment by mx-moth** [comment]
Fixed in #62
